code/game/objects/structures/props/beam_prism.dm

```diff
@@ -0,0 +1,218 @@
+		if(new_bearing <= -1 || new_bearing > 360)
+			to_chat(user, "<span class='warning'>Rotating \the [src] [new_bearing] degrees would be a waste of time.</span>")
+			return
+	else
+		var/choice = input("What point do you want to set \the [src] to?", "[name]") as null|anything in compass_directions
+		new_bearing = round(compass_directions[choice])
+
+	var/rotate_degrees = new_bearing - degrees_from_north
+
+	if(new_bearing == 360) // Weird artifact.
+		new_bearing = 0
+	degrees_from_north = new_bearing
+
+	var/two_stage = 0
+	if(rotate_degrees == 180 || rotate_degrees == -180)
+		two_stage = 1
+		var/multiplier = pick(-1, 1)
+		rotate_degrees = multiplier * (rotate_degrees / 2)
+
+	playsound(src, interaction_sound, 50, 1)
+	if(two_stage)
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 3)
+		sleep(3)
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 3)
+	else
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 6) //Can't update transform because it will reset the angle.
+
+/obj/structure/prop/prism/proc/rotate_auto(var/new_bearing)
+	if(rotation_lock)
+		visible_message("<span class='notice'>\The [src] shudders.</span>")
+		playsound(src, 'sound/effects/clang.ogg', 50, 1)
+		return
+
+	visible_message("<span class='notice'>\The [src] rotates to a bearing of [new_bearing].</span>")
+
+	var/rotate_degrees = new_bearing - degrees_from_north
+
+	if(new_bearing == 360)
+		new_bearing = 0
+	degrees_from_north = new_bearing
+
+	var/two_stage = 0
+	if(rotate_degrees == 180 || rotate_degrees == -180)
+		two_stage = 1
+		var/multiplier = pick(-1, 1)
+		rotate_degrees = multiplier * (rotate_degrees / 2)
+
+	playsound(src, interaction_sound, 50, 1)
+	if(two_stage)
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 3)
+		sleep(3)
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 3)
+	else
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 6)
+
+/obj/structure/prop/prism/bullet_act(var/obj/item/projectile/Proj)
+	if(istype(Proj, redirect_type))
+		visible_message("<span class='danger'>\The [src] redirects \the [Proj]!</span>")
+		flick("[initial(icon_state)]+glow", src)
+
+		var/new_x = (1 * round(10 * cos(degrees_from_north - 90))) + x //Vectors vectors vectors.
+		var/new_y = (-1 * round(10 * sin(degrees_from_north - 90))) + y
+		var/turf/curloc = get_turf(src)
+
+		Proj.penetrating += 1 // Needed for the beam to get out of the turret.
+
+		Proj.redirect(new_x, new_y, curloc, null)
+
+/obj/structure/prop/prism/incremental
+	free_rotate = 0
+	description_info = "This device is capable of redirecting any beam projectile, but only locks to specific positions in rotation."
+
+/obj/structure/prop/prism/incremental/externalcont
+	external_control_lock = 1
+	description_info = "This device is capable of redirecting any beam projectile, but only locks to specific positions in rotation when rotated by a control dial."
```

Comment:
`but can only be rotated by a control dial to specific positions.`
Current text implies it can be rotated to any position if you don't use the dial